dev/buildtool/source_code_manager.py

```diff
@@ -0,0 +1,294 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Responsible for managing the source code that we are building.
+
+This includes fetching the source to be built as well as manipulating
+the git repositories for tagging and annotations.
+"""
+
+# Ideally something like a Pool is better because it will run in
+# different processes to bypass some of the global interpreter lock.
+# However because we often run commands in these, which forks another process,
+# we seem to deadlock sometimes (always when running gradle). I think this
+# could be because this pool fork has some locks grabbed which leads to
+# deadlock when something is trying to grab a lock that was already locked
+# at the point of the first fork. This doesnt really make sense because it
+# only happens with gradle and not with git, and the gradle jobs do complete
+# in the pool'd thread so at that point there shouldnt be a difference between
+# them. So maybe something else is going on in the build command.
+
+from multiprocessing.pool import (
+    Pool,
+    ThreadPool)
+
+import logging
+import os
+import yaml
+
+# pylint: disable=relative-import
```

Comment:
If pylint doesn't like relative imports, what is the proper 'pythonic' way to handle separate modules? Different published libraries? Just curious, I'm fine with the relative imports we have.

Reply:
It wants a leading "."
Or maybe I could just change my pylint.rc file.